language-server/src/main/scala/meta/languageserver/ScalametaLanguageServer.scala

```diff
@@ -1,45 +1,52 @@
-
-import langserver.core.{LanguageServer, TextDocument, MessageReader}
+import java.nio.file.Paths
+import scalafix.languageserver.ScalafixResult
+import scalafix.languageserver.ScalafixService
+import langserver.core.LanguageServer
 import langserver.messages._
 import langserver.types._
+import org.langmeta.io.AbsolutePath
 
-class ScalametaLanguageServer(in: InputStream, out: OutputStream) extends LanguageServer(in, out) {
+class ScalametaLanguageServer(cwd: AbsolutePath,
+                              in: InputStream,
+                              out: OutputStream)
+    extends LanguageServer(in, out) {
+  val ps = new PrintStream(out)
+  val scalafixService = new ScalafixService(cwd, out, connection)
 
-  override def initialize(pid: Long, rootPath: String, capabilities: ClientCapabilities): ServerCapabilities = {
-    logger.info(s"Initialized with $pid, $rootPath, $capabilities")
+  override def initialize(
+      pid: Long,
+      rootPath: String,
+      capabilities: ClientCapabilities): ServerCapabilities = {
+    logger.info(s"Initialized with $cwd, $pid, $rootPath, $capabilities")
 
     ServerCapabilities(completionProvider = None)
   }
 
-  private[this] def runScalafixLint(uri: String): List[Diagnostic] = {
-    // TODO(gabro): here's where we have to invoke scalafix and collect all the linter errors/warnings
-
-    // Dummy example of Diagnostic
-    List(Diagnostic(
-      range = Range(start = Position(line = 5, character = 0), end = Position(line = 5, character = 6)),
-      severity = Some(DiagnosticSeverity.Error),
-      code = None,
-      source = Some("scalafix"),
-      message = "Hey, this is not good"
-    ))
-
-  }
-
-  override def onChangeTextDocument(td: VersionedTextDocumentIdentifier, changes: Seq[TextDocumentContentChangeEvent]): Unit = {
-    val diagnostics = runScalafixLint(td.uri)
-    connection.publishDiagnostics(td.uri, diagnostics)
+  def report(result: ScalafixResult): Unit = {
+    // NOTE(olafur) I must be doing something wrong, but nio.Path.toUri.toString
+    // produces file:/// instead of file:/ which is what vscode expects
+    val fileUri = s"file:${cwd.resolve(result.path)}"
+    connection.publishDiagnostics(fileUri, result.diagnostics)
   }
 
-  override def onChangeWatchedFiles(changes: Seq[FileEvent]): Unit = changes match {
-    case FileEvent(uri, FileChangeType.Created | FileChangeType.Changed | FileChangeType.Deleted) +: _ =>
-      // TODO(gabro): not sure what is the strategy to manage .semanticdb is going to be
-      // but just in case we're being notified here about changes
-      logger.info(s"$uri changed, created or deleted")
-    case _ => ()
-  }
+  override def onChangeWatchedFiles(changes: Seq[FileEvent]): Unit =
+    changes match {
+      case FileEvent(uri, FileChangeType.Created | FileChangeType.Changed) +: _ =>
+        val semanticdbUri = Paths.get(new URI(uri))
+        // TODO(gabro): not sure what is the strategy to manage .semanticdb is going to be
```

Comment:
this TODO can go away. Also I we will use this method also to watch the changes to the `.scalafix.conf` file, so I would check the file extension before assuming it's a `.semanticdb`.